samtranslator/model/apigatewayv2.py

```diff
@@ -230,31 +223,35 @@ def generate_openapi(self) -> Dict[str, Any]:
                 ),
                 {"__FunctionArn__": self.function_arn},
             )
-            openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerUri"] = authorizer_uri  # type: ignore[index]
+            openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerUri"] = authorizer_uri
 
             # Set authorizerCredentials if present
             function_invoke_role = self._get_function_invoke_role()  # type: ignore[no-untyped-call]
             if function_invoke_role:
-                openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerCredentials"] = function_invoke_role  # type: ignore[index]
-
-            # Set authorizerResultTtlInSeconds if present
-            reauthorize_every = self._get_reauthorize_every()  # type: ignore[no-untyped-call]
-            if reauthorize_every is not None:
-                openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerResultTtlInSeconds"] = reauthorize_every  # type: ignore[index]
+                openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerCredentials"] = function_invoke_role
 
             # Set identitySource if present
             if self.identity:
-                openapi[APIGATEWAY_AUTHORIZER_KEY]["identitySource"] = self._get_identity_source()  # type: ignore[no-untyped-call, index]
+                sam_expect(self.identity, self.api_logical_id, f"Auth.Authorizers.{self.name}.Identity").to_be_a_map()
+                # Set authorizerResultTtlInSeconds if present
+                reauthorize_every = self.identity.get("ReauthorizeEvery")
+                if reauthorize_every is not None:
+                    openapi[APIGATEWAY_AUTHORIZER_KEY]["authorizerResultTtlInSeconds"] = reauthorize_every
+
+                # Set identitySource if present
+                openapi[APIGATEWAY_AUTHORIZER_KEY]["identitySource"] = self._get_identity_source(self.identity)
 
             # Set authorizerPayloadFormatVersion. It's a required parameter
-            openapi[APIGATEWAY_AUTHORIZER_KEY][  # type: ignore[index]
+            openapi[APIGATEWAY_AUTHORIZER_KEY][
                 "authorizerPayloadFormatVersion"
             ] = self.authorizer_payload_format_version
 
             # Set authorizerPayloadFormatVersion. It's a required parameter
```

Comment:
This comment seems wrong

Reply:
yeah it feels like a copy-pasta, will correct it in next rev